frontend/src/components/elements/DataFrame/DataFrame.tsx

```diff
@@ -197,6 +204,14 @@ export function DataFrame({
   // widths.
   recomputeSizeIfNeeded()
 
+  useEffect(() => {
+    if (sortColumn - headerCols >= nCols) {
```

Comment:
Totally also outside of the scope of this PR as well but headerCols threw me off for the longest time. Reading through, I thought it was the number of header columns which made no sense to me how this would work. I had to actually debug this to understand what this was. Perhaps it could benefit from better comments but not really scope of this PR. Just wanted to point it out

Reply:
The whole file is a big mess, and there are tons of issues. I didn't bother changing it now as I'm going to revamp it during the Arrow refactor.